app/controllers/concerns/application_activities_concern.rb

```diff
@@ -0,0 +1,9 @@
+module ApplicationActivitiesConcern
+  extend ActiveSupport::Concern
+
+  def recent_activities(course, number)
+    return if number < 0 || !number.is_a?(Integer)
+    PublicActivity::Activity.where(owner: course).where.not('key like ?', '%notification').
```

Comment:
So you have three fields in a notification:

 - Trackable
 - Owner
 - Recipient

How do you know which field is used for what? How is your design consistent?